src/Component/Form/Type/VariationChoiceType.php

```diff
@@ -49,13 +49,23 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         $choices = $this->pool->getProvider($options['product'])->getVariationsChoices($options['product'], $options['fields']);
 
         foreach ($choices as $choiceTitle => $choiceValues) {
+            $choiceOptions = array(
+                'label' => sprintf('form_%s', $choiceTitle),
+                'translation_domain' => 'SonataProductBundle',
+            );
+            // NEXT_MAJOR: Remove this "if" (when requirement of Symfony is >= 2.7)
+            if (method_exists('Symfony\Component\Form\AbstractType', 'configureOptions')) {
+                $choiceValues = array_flip($choiceValues);
+                // choice_as_value option is not needed in SF 3.0+
+                if (method_exists('Symfony\Component\Form\FormTypeInterface', 'setDefaultOptions')) {
+                    $choiceOptions['choices_as_values'] = true;
+                }
+            }
+            $choiceOptions['choices'] = $choiceValues;
+
             $builder->add($choiceTitle, $choiceType, array_merge(
-                    array('translation_domain' => 'SonataProductBundle'),
```

Comment:
I'll move `$choiceOptions` before `$options['field_options']` in `array_merge` then it should be okay.